includes/compat/blocks/class-blocks-data-extractor.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Class Blocks_Data_Extractor
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay;
+
+use Automattic\WooCommerce\Blocks\Integrations\IntegrationRegistry;
+
+defined( 'ABSPATH' ) || exit; // block direct access.
+
+/**
+ * Extract data fields from certain block based plugins.
+ */
+class Blocks_Data_Extractor {
+
+	/**
+	 * Instance of the integration registry.
+	 *
+	 * @var IntegrationRegistry
+	 */
+	private $integration_registry;
+
+	/**
+	 * An array of blocks to extract data fields.
+	 *
+	 * @var array
+	 */
+	private $blocks = [];
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->integration_registry = new IntegrationRegistry();
+		$this->blocks               = $this->get_available_blocks();
+
+		$this->register_blocks();
```

Comment:
Two questions here:

1. Is there any advantage in running this code in the constructor, and then retrieving the data with `get_data` at some point later on, compared to doing it all together in the same call? It's all working well this way. I just think it makes it harder to follow how things play together.
2.  Should we unregister blocks after we get their data? I'm not sure if there are side effects of not doing it, maybe we could check with Nadir. In this context, I get this is only called in an ajax call and I think it's unlikely there are any.

Reply:
Because we are using a separate `integration_registryy` I didn't see a reason to unregister. I was also not sure about the added overhead of doing everything in `get_data`. But having looked at it again following your comment, the methods we are calling in `integration_registry` seems to be pretty light, so moving that logic to `get_data` makes sense just to be cautious. I've done this in https://github.com/Automattic/woocommerce-payments/pull/5483/commits/714c44c7c0f0fc1baf64c088a527279929cb10ff. Thanks!

Still instantiating `integration_registry` under the constructor to avoid it being passed back and forth.

Reply:
Makes sense! Thanks Malith!